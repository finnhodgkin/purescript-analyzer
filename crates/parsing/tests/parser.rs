macro_rules! parser_tests {
    ($($name:ident => $source:expr),+ $(,)?) => {
        $(
            #[test]
            fn $name() {
                let lexed = lexing::lex($source);
                let tokens = lexing::layout(&lexed);
                let (node, errors) = parsing::parse(&lexed, &tokens);
                insta::assert_debug_snapshot!((node, errors));
            }
        )+
    };
}

parser_tests!(
    comment_pre_post => include_str!("parser/CommentPrePost.purs"),
    lexer_error_inserted => include_str!("parser/LexerErrorInserted.purs"),
    module_export => include_str!("parser/ModuleExport.purs"),
    module_export_class => include_str!("parser/ModuleExportClass.purs"),
    module_export_class_error_end => include_str!("parser/ModuleExportClassErrorEnd.purs"),
    module_export_comma_error => include_str!("parser/ModuleExportCommaError.purs"),
    module_export_data => include_str!("parser/ModuleExportData.purs"),
    module_export_data_comma_error => include_str!("parser/ModuleExportDataCommaError.purs"),
    module_export_data_item_after_double_period => include_str!("parser/ModuleExportDataItemAfterDoublePeriod.purs"),
    module_export_data_item_before_double_period => include_str!("parser/ModuleExportDataItemBeforeDoublePeriod.purs"),
    module_export_data_item_invalid => include_str!("parser/ModuleExportDataItemInvalid.purs"),
    module_export_empty => include_str!("parser/ModuleExportEmpty.purs"),
    module_export_invalid => include_str!("parser/ModuleExportInvalid.purs"),
    module_export_type_operator => include_str!("parser/ModuleExportTypeOperator.purs"),
    module_export_type_operator_error => include_str!("parser/ModuleExportTypeOperatorError.purs"),
    module_export_type_operator_error_end => include_str!("parser/ModuleExportTypeOperatorErrorEnd.purs"),
    module_header => include_str!("parser/ModuleHeader.purs"),
    module_header_prefixed => include_str!("parser/ModuleHeaderPrefixed.purs"),
);
