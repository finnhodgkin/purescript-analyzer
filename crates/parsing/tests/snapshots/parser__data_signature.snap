---
source: crates/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..55
      ModuleHeader@0..28
        MODULE@0..6 "module"
        Comment@6..7
          WHITESPACE@6..7 " "
        ModuleName@7..21
          UPPER@7..20 "DataSignature"
          Comment@20..21
            WHITESPACE@20..21 " "
        ModuleExportList@21..21
        WHERE@21..26 "where"
        Comment@26..28
          WHITESPACE@26..28 "\n\n"
      LAYOUT_START@28..28 ""
      ModuleImports@28..28
      ModuleStatements@28..55
        DataSignature@28..55
          DATA@28..32 "data"
          Comment@32..33
            WHITESPACE@32..33 " "
          UPPER@33..38 "Maybe"
          Comment@38..39
            WHITESPACE@38..39 " "
          DOUBLE_COLON@39..41 "::"
          Comment@41..42
            WHITESPACE@41..42 " "
          TypeArrow@42..55
            TypeConstructor@42..47
              QualifiedName@42..47
                UPPER@42..46 "Type"
                Comment@46..47
                  WHITESPACE@46..47 " "
            RIGHT_ARROW@47..49 "->"
            Comment@49..50
              WHITESPACE@49..50 " "
            TypeConstructor@50..55
              QualifiedName@50..55
                UPPER@50..54 "Type"
                Comment@54..55
                  WHITESPACE@54..55 "\n"
      LAYOUT_END@55..55 ""
      END_OF_FILE@55..55 ""
    ,
    [],
)
